arch/Kconfig

```diff
@@ -498,7 +502,7 @@ config ARCH_HAVE_PERF_EVENTS
 
 config ARCH_PERF_EVENTS
 	bool "Configure hardware performance counting"
-	default y
+	default y if ARCH_NEED_PERF
```

Comment:
why need a new option? if we already have ARCH_HAVE_PERF_EVENTS and ARCH_PERF_EVENTS.R

Reply:
`ARCH_HAVE_PERF_EVENTS` informs whether arch supports `ARCH_PERF_EVENTS`.  `ARCH_PERF_EVENTS` enables arch specific `up_perf_xxx`. But according to c928acc9ffed574b139d7508a1338ce71d772291 you can have `ARCH_PERF_EVENTS` supported in arch but still use a different `up_perf_xxx` implementation. 
So the currently available flags serve a different purpose than what is needed here.
We need here a flag that is set when any system component needs perf.

Reply:
https://github.com/apache/nuttx/pull/10834 will decouple system component from up_perf.